CIP-0035/CIP-0035.md

```diff
@@ -0,0 +1,218 @@
+---
+CIP: 35
+Title: On-Chain Token Metadata Standard
+Authors: Matt Ho <matt.ho@gmail.com>
+Comments-URI: https://github.com/cardano-foundation/CIPs/pull/137
+Status: Draft
+Type: Standards
+Created: 2020-10-12
+License: CC-BY-4.0
+---
+
+# Abstract
+
+This specification defines a new transaction metadata label (i.e. `20`) and entry for [CIP-0031] to associate token metadata 
+during a minting transaction.  Unlike CIP-28, this proposal is intended to define a specific schema to allow basic information
+about fungible tokens to be provided On-Chain.
+
+`20` is proposed as the metadata label in homage to the `ERC20` standard on Ethereum.
+
+# Motivation
+
+The (Cardano Token Registry)[https://github.com/cardano-foundation/cardano-token-registry] is a centralized Off-Chain governance system to manage token 
+metadata.  This proposal aims to provide an On-Chain solution to the same problems that will afford users a much higher degree of trust than the existing
+system.  
+
+Specifically, this proposal aims to solve for the following:
+
+* Provide a standardized way to associate an icon and a ticker with a token
+* Provide a mechanism that allows the viewer to evaluate the trustworthiness and/or authenticity of a token
+
+# Specification
+
+Minters of tokens on the Cardano blockchain main optionally choose to associate the following transaction metadata to facilitate off-chain labeling of tokens.
+When doing so, transaction metadata MUST be included in the same transaction that mints the asset id.
```

Comment:
This does not precise which mint tx to use when there are several for the same asset.

CIP 25 requires to use the last mint tx (burns are not considered mints but this should be precised), which allows to update the metadata as long as the policy allows it by minting a duplicate then burning it.

This point should be precised in my opinion (for now I don't know how to implement the lookup).

Reply:
I agree that it needs to be precise. I recommend that the FIRST time the "20" metadata appears is the only one that matters. For something like decimals, you should get it right the first time and never change it.